### ISSUE TITLE
Improve debug console display

### ISSUE THREAD

**Issue by CameronBieganek** [julia-vscode/julia-vscode#1106]
Is there any way to improve the printing in the debug console? Here's an example which is rather difficult to read:

![image](https://user-images.githubusercontent.com/8310743/79079776-c51bcb00-7cd6-11ea-80a0-a0ee69643c99.png)

There's not much contrast. I think the other problem in this example is that the short-form printing of the nested arrays is pretty difficult to read. These particular arrays print much nicer in the regular REPL. I apologize for the noise if this isn't really possible to improve with the VS Code debugger.

**Comment by davidanthoff** [comment]
Yes, we can definitely improve this!

Note to myself: the DAP message that is responsible for this is `evaluate`. It covers both the debug console and things like the watch window. Right now I'm just using `repr` in all of these cases [here](https://github.com/julia-vscode/julia-vscode/blob/f2d186aa866df89ae27b64134e8927db48bf7f51/scripts/debugger/debugger_requests.jl#L611). The protocol does allow me to distinguish in what context the evaluation is happening, so we just need to pass that info on in our own protocol, and then call a more useful display method on the Julia side for the REPL case.

**Comment by phlavenk** [comment]
It is true that the readability is not that great. What i like is that e.g array output in the debug console can be directly copy-pasted as valid array definition. Maybe i'm just misusing this in my workflow, where I step deep into call-stack  and capture variables used for calling a function I'd like to isolate and debug / test / perf-tune the function in isolation. But anyhow, i can just call `repr(val)` if the defaults change...

**Comment by RGerzaguet** [comment]
Hello there,
Sorry to update this aggressively but I also wondered if it is possible to have more information when using the debugger console, especially when errors in the current evaluation occur.
For instance, if in the debug console you run the following erroneous syntax, we get this 
```
1 + [1;2]
#error
```
Any idea on how we can get the regular stack trace ?
```
julia> 1+[1;2]
ERROR: MethodError: no method matching +(::Int64, ::Array{Int64,1})
For element-wise addition, use broadcasting with dot syntax: scalar .+ array
Closest candidates are:
  +(::Any, ::Any, ::Any, ::Any...) at operators.jl:538
  +(::T, ::T) where T<:Union{Int128, Int16, Int32, Int64, Int8, UInt128, UInt16, UInt32, UInt64, UInt8} at int.jl:86
  +(::Union{Int16, Int32, Int64, Int8}, ::BigInt) at gmp.jl:531
  ...
Stacktrace:
 [1] top-level scope at REPL[3]:1
```

(Note, I often use the debug console to explore additional code, and I often do mistakes :D)

**Comment by fleimgruber** [comment]
@davidanthoff I hit this today and was curious whether #1838 and #1796 might be related as there were some upstream changes regarding DAP in https://github.com/microsoft/vscode/issues/113921.

**Comment by pfitzseb** [comment]
No, those are unrelated (only relevant for errors happening while debugging, not in the debug console). We can just fix this issue without any upstream changes, afaict.

**Comment by fleimgruber** [comment]
@pfitzseb thanks for checking back and clarifying. I was only thinking that the upstream changes may affect or help this issue in any way.

**Comment by imamrb** [comment]
Is there any update on this? 🤔 It would be great to see some improvements in the debug console. Right now, the command output isn’t highlighted, and I can’t expand the output hash in a readable format. Instead, I only get expandable views of the underlying data structures and metadata, which aren’t really needed as frequently.


![Image](https://github.com/user-attachments/assets/73bc212d-89f2-450b-8bad-c0cc6b2cbea8)
